test/functional/tests/lazy_writes/test_lazy_writes_signals.py

```diff
@@ -0,0 +1,399 @@
+          Test if OpenCAS transmits FLUSH and FUA signals to core device in modes with lazy writes.
+        pass_criteria:
+          - FLUSH requests should be passed to core device.
+          - FUA requests should be passed to core device.
+    """
+    with TestRun.step("Set mark in syslog to not read entries existing before the test."):
+        Logs._read_syslog(Logs.last_read_line)
+
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(2, Unit.GibiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.scsi_debug_devices[0]
+
+    with TestRun.step("Start cache and add SCSI device with xfs filesystem as core."):
+        cache = casadm.start_cache(cache_part, cache_mode)
+        core_dev.create_filesystem(Filesystem.xfs)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Mount exported object."):
+        if core.is_mounted():
+            core.unmount()
+        core.mount(mount_point)
+
+    with TestRun.step("Turn off cleaning policy."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step("Create temporary file on exported object."):
+        tmp_file = create_random_test_file(f"{mount_point}/tmp.file", Size(1, Unit.GibiByte))
+        os_utils.sync()
+
+    with TestRun.step("Flush cache."):
+        cache.flush_cache()
+        os_utils.sync()
+
+    with TestRun.step(f"Check {syslog_path} for flush request and delete temporary file."):
+        Logs.check_syslog_for_signals()
+        tmp_file.remove(True)
+
+    with TestRun.step("Create temporary file on exported object."):
+        tmp_file = create_random_test_file(f"{mount_point}/tmp.file", Size(1, Unit.GibiByte))
+        os_utils.sync()
+
+    with TestRun.step("Flush core."):
+        core.flush_core()
+        os_utils.sync()
+
+    with TestRun.step(f"Check {syslog_path} for flush request and delete temporary file."):
+        Logs.check_syslog_for_signals()
+        tmp_file.remove(True)
+
+    with TestRun.step("Turn on alru cleaning policy and set policy params."):
+        cache.set_cleaning_policy(CleaningPolicy.alru)
+        cache.set_params_alru(FlushParametersAlru(
+            Time(milliseconds=5000), 10000, Time(seconds=10), Time(seconds=10))
+        )
+
+    with TestRun.step("Create big temporary file on exported object."):
+        tmp_file = create_random_test_file(f"{mount_point}/tmp.file", Size(5, Unit.GibiByte))
+        os_utils.sync()
+
+    with TestRun.step("Wait for automatic flush from alru cleaning policy and check log."):
+        wait_time = (
+            int(cache.get_flush_parameters_alru().staleness_time.total_seconds())
+            + int(cache.get_flush_parameters_alru().activity_threshold.total_seconds())
+            + int(cache.get_flush_parameters_alru().wake_up_time.total_seconds())
+            + 5
+        )
+        sleep(wait_time)
+        os_utils.sync()
```

Comment:
Not sure if this sync here won't break the behavior under test - we're waiting for flush that comes from cache cleaning, not the one triggered by `os_utils.sync()`